content/how-to/long-term-caching.md

```diff
@@ -0,0 +1,258 @@
+          JSON.stringify(stats.toJson()));
+      });
+    }
+  ]
+};
+```
+
+Alternatively, just use one of these plugins to export JSON files:
+
+* https://www.npmjs.com/package/webpack-manifest-plugin
+* https://www.npmjs.com/package/assets-webpack-plugin
+
+A sample output of webpack-manifest-plugin for our config looks like:
+
+```
+{
+  “main.js”: “main.155567618f4367cd1cb8.js”,
+  “vendor.js”: “vendor.c2330c22cd2decb5da5a.js”
+}
+```
+
+The rest depends on your server setup. There is a nice [walk through for Rails-based projects](http://clarkdave.net/2015/01/how-to-use-webpack-with-rails/#including-precompiled-assets-in-views). Or, if your application doesn’t rely on any server-side rendering, it’s often enough to generate a single `index.html` file for your application. To do so, just use these 2 amazing plugins:
+
+* https://github.com/ampedandwired/html-webpack-plugin
+* https://github.com/szrenwei/inline-manifest-webpack-plugin
+
+It will simplify the setup dramatically.
+
+We’re done, you might think. Well, almost.
+
+## Deterministic hashes
+
+To minimise the size of generated files, webpack uses identifiers instead of module names. During compilation identifiers are generated, mapped to chunk filenames and then put into a JavaScript object called *chunk manifest*. It is (along with some bootstrapping code) then placed into the entry chunk and it is crucial for webpack-packaged code to work.
+
+The problem with this is the same as before: whenever we change any part of the code, it will, even if the rest of its contents wasn’t altered, update our entry chunk to include the new manifest. This, in turn, will lead to a new hash and dismiss the long-term caching.
+
+To fix that, we should use [chunk-manifest-webpack-plugin](https://github.com/diurnalist/chunk-manifest-webpack-plugin) which will extract that manifest to a separate JSON file. Here is an updated webpack.config.js which will produce chunk-manifest.json in our build directory:
+
+```js
+// webpack.config.js
+var ChunkManifestPlugin = require('chunk-manifest-webpack-plugin');
+
+module.exports = {
+  // your config values here
+  plugins: [
+    new ChunkManifestPlugin({
+      filename: "chunk-manifest.json",
+      manifestVariable: "webpackManifest"
+    })
+  ]
+};
+```
+
+Since we removed the manifest from entry chunk, now it’s our responsibility to provide webpack with it. You have probably noticed the `manifestVariable` option in the example above. This is the name of the global variable where webpack will look for the manifest JSON and this is why *it should be defined before we require our bundle in HTML*. This is easy as inlining the contents of the JSON in HTML. Our HTML head section should look like this:
+
+```html
+<html>
+  <head>
+    <script>
+    //<![CDATA[
+    window.webpackManifest = {"0":"main.3d038f325b02fdee5724.js","1":"1.c4116058de00860e5aa8.js"}
+    //]]>
+    </script>
+  </head>
+  <body>
+  </body>
+</html>
+```
```

Comment:
I'm already mentioning it here https://github.com/webpack/webpack.io/pull/75/files/c4d390c86024d9f7263bfca2e2fada9764a89b10#diff-c4fb0061ead1aff0a7a16748012e2e33R218 and https://github.com/webpack/webpack.io/pull/75/files/c4d390c86024d9f7263bfca2e2fada9764a89b10#diff-c4fb0061ead1aff0a7a16748012e2e33R137 but will add another tip.